### ISSUE TITLE
UI: Support viewing logs for containers created by resource template

### ISSUE THREAD

**Issue by mirocody** [argoproj/argo-workflows#7012]
# Summary

AFAIK, currently argo can only log container defined in Workflow step, not the container it creates through resource template. It will be useful if users can specify the container they want to log.

# Use Cases

For example, if users want to use argo and kubeflow training operator for ML training, they can use the follow template 
```yaml
  - metadata:
      labels:
        user: xxx
    name: step4
    resource:
      action: create
      failureCondition: failed in (status.replicaStatuses.Master, status.replicaStatuses.Worker)
      manifest: |
        apiVersion: kubeflow.org/v1
        kind: PyTorchJob
        metadata:
          generateName: step4-pytorchjob-
          labels:
            user: xxx
          namespace: xxx
        spec:
           <pytorchjob related spec>
```
However, currently argo can only log the container of that "argo container", which looks like below:
```
time="2021-10-21T07:01:06.195Z" level=info msg="0/1 success conditions matched"
time="2021-10-21T07:01:06.195Z" level=info msg="Waiting for resource pytorchjob.kubeflow.org/pytorchjob-l8kc9 in namespace argo resulted in retryable error: Neither success condition nor the failure condition has been matched. Retrying..."
time="2021-10-21T07:01:11.199Z" level=info msg="Get pytorchjobs 200"
time="2021-10-21T07:01:11.199Z" level=info msg="failure condition '{failed in [status.replicaStatuses.Master status.replicaStatuses.Worker]}' evaluated false"
time="2021-10-21T07:01:11.199Z" level=info msg="success condition '{status.replicaStatuses.Master.succeeded exists []}' evaluated false"
time="2021-10-21T07:01:11.199Z" level=info msg="0/1 success
```
Can we support this feature? Thanks.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by berlinsaint** [comment]
i am not familar with ts

**Comment by wjhhuizi** [comment]
Providing an example to extract resource log in parallel step. Hope this can help you.

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: sparkjob-
  annotations:
    pipelines.kubeflow.org/kfp_sdk_version: 1.8.22
  labels: 
    pipelines.kubeflow.org/kfp_sdk_version: 1.8.22
spec:
  serviceAccountName: pipeline-runner
  entrypoint: sparkjob
  templates:
  - name: sparkjob
    steps:
    - - name: sparkjob-run
        template: sparkjob-run
      - name: sparkjob-driver-log
        template: sparkjob-driver-log

  - name: sparkjob-run
    resource:
      action: create
      successCondition: status.applicationState.state == COMPLETED
      failureCondition: status.applicationState.state == FAILED
      manifest: |
        apiVersion: "sparkoperator.k8s.io/v1beta2"
        kind: SparkApplication
        metadata:
          name: {{workflow.name}}
          namespace: spark-operator-35
        spec:
          type: Python
          pythonVersion: "3"
          mode: cluster
          image: "your/spark/image:latest"
          imagePullPolicy: Always
          mainApplicationFile: local:///opt/spark/work-dir/app.py
          sparkVersion: "3.5.1"
          sparkConf:
            spark.shuffle.io.retryWait: "3s"
            spark.network.timeout: "800"
          restartPolicy:
            type: OnFailure
            onFailureRetries: 1
            onFailureRetryInterval: 10
            onSubmissionFailureRetries: 1
            onSubmissionFailureRetryInterval: 20
          driver:
            annotations:
              sidecar.istio.io/inject: "false"
            securityContext:
              runAsUser: 0
            cores: 1
            coreLimit: "1200m"
            memory: "64G"
            labels:
              version: 3.5.1
              workflows.argoproj.io/workflow: "{{workflow.name}}"
            serviceAccount: spark-operator-spark
          executor:
            annotations:
              sidecar.istio.io/inject: "false"
            securityContext:
              runAsUser: 0
            cores: 3
            instances: 1
            memory: "32G"
            labels:
              version: 3.5.1
              workflows.argoproj.io/workflow: "{{workflow.name}}"

  - name: sparkjob-driver-log
    container:
      image: bitnami/kubectl:1.27
      command:
      - bash
      - -c
      args:
      - |
        POD_NAME="{{workflow.name}}-driver"
        NAMESPACE="{{workflow.namespace}}"
        while true; do
          POD_STATUS=$(kubectl get pod ${POD_NAME} -n ${NAMESPACE} --ignore-not-found)
          if [ -n "$POD_STATUS" ]; then
            echo "Pod $POD_NAME is created..."
            break
          else
            echo "Waiting for pod $POD_NAME to be created..."
            sleep 1
          fi
        done && kubectl wait pod {{workflow.name}}-driver -n {{workflow.namespace}} --for=condition=ready --timeout=600s && kubectl logs {{workflow.name}}-driver -n {{workflow.namespace}} --follow
```

**Comment by agilgur5** [comment]
> See #7067

This was resolved by #7067, so this issue is effectively a duplicate of #6910

**Comment by agilgur5** [comment]
> This was resolved by #7067

Or that didn't quite add a UI feature for it per se or document how to use it for this purpose

**Comment by agilgur5** [comment]
> Or that didn't quite add a UI feature for it

https://github.com/argoproj/argo-workflows/pull/10397 added the UI feature as I understand, although it is still not documented
